compiler/language_server/src/features_ir.rs

```diff
@@ -55,13 +49,8 @@ pub struct ViewIrParams {
 
 impl Server {
     pub async fn candy_view_ir(&self, params: ViewIrParams) -> jsonrpc::Result<String> {
-        let project_directory = {
-            let state = self.state.read().await;
-            state.require_running().project_directory.to_owned()
-        };
-        let config = IrConfig::decode(&params.uri, project_directory);
-
         let state = self.state.read().await;
+        let config = IrConfig::decode(&params.uri, &state.require_running().packages_path);
```

Comment:
```suggestion
        let config = IrConfig::decode(&params.uri, &state.require_running().packages_path);
        let state = self.state.read().await;
```

(Swap the lines to keep the lock for a shorter time)